routers/web/websocket/websocket.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package websocket
+
+import (
+	"time"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/web"
+
+	"github.com/olahol/melody"
+)
+
+var m *melody.Melody
+
+func Init(r *web.Route) {
+	m = melody.New()
+	r.Any("/ui-updates", WebSocket)
```

Comment:
Make it just `/ws` please. There must only be a single websocket endpoint, which is also important as some reverse proxies will want to match websocket by URL path.

Reply:
I guess it should be `/-/ws` to avoid reserving a new user name?

Reply:
Right, good idea. `/api/ws` may also be a idea, but I guess I prefer yours.

Reply:
renamed it to `/-/ws`